pkg/maglev/maglev.go

```diff
@@ -4,80 +4,141 @@
 package maglev
 
 import (
+	"cmp"
 	"context"
 	"encoding/base64"
 	"fmt"
+	"iter"
 	"runtime"
 	"slices"
+	"strconv"
+	"strings"
 
+	"github.com/cilium/hive/cell"
 	"github.com/cilium/workerpool"
-	"github.com/mackerelio/go-osstat/memory"
+	"github.com/spf13/pflag"
 
 	"github.com/cilium/cilium/pkg/loadbalancer"
+	"github.com/cilium/cilium/pkg/lock"
 	"github.com/cilium/cilium/pkg/murmur3"
 )
 
+var Cell = cell.Module(
+	"maglev",
+	"Maglev table computations",
+
+	cell.Config(DefaultConfig),
+	cell.Provide(New),
+)
+
 const (
 	DefaultTableSize = 16381
 
 	// seed=$(head -c12 /dev/urandom | base64 -w0)
 	DefaultHashSeed = "JLfvgnHc2kaSUFaI"
+
+	MaglevTableSizeName = "bpf-lb-maglev-table-size"
+	MaglevHashSeedName  = "bpf-lb-maglev-hash-seed"
 )
 
-var (
-	seedMurmur uint32
+type Config struct {
+	// Maglev backend table size (M) per service. Must be prime number.
+	MaglevTableSize uint `mapstructure:"bpf-lb-maglev-table-size"`
+
+	// MaglevHashSeed contains the cluster-wide seed for the hash(es).
+	MaglevHashSeed string `mapstructure:"bpf-lb-maglev-hash-seed"`
+}
+
+func (def Config) Flags(flags *pflag.FlagSet) {
+	flags.Uint(MaglevTableSizeName, def.MaglevTableSize, "Maglev per service backend table size (parameter M)")
```

Comment:
Q: Should we list the set of possible primes in the cmd ref ?

Reply:
Yep, added!